From ecfe686834ff2ff24343667c1b44d204e2c13ff7 Mon Sep 17 00:00:00 2001
From: penglezos <panagiotisegl@gmail.com>
Date: Sat, 17 Feb 2024 18:56:29 +0200
Subject: [PATCH 1/8] LocalImageResolver: Stop the spam

Spaming in logcat:
2514  2591 D LocalImageResolver: Couldn't use ImageDecoder for drawable, falling back to non-resized load.

Change-Id: I24dbb83070a6061e15f2f7cc777387cd65083afa
Signed-off-by: penglezos <panagiotisegl@gmail.com>
---
 .../java/com/android/internal/widget/LocalImageResolver.java | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/core/java/com/android/internal/widget/LocalImageResolver.java b/core/java/com/android/internal/widget/LocalImageResolver.java
index 9ef7ce38fc09..62fdc4785be0 100644
--- a/core/java/com/android/internal/widget/LocalImageResolver.java
+++ b/core/java/com/android/internal/widget/LocalImageResolver.java
@@ -38,6 +38,7 @@ import java.io.IOException;
 /** A class to extract Drawables from a MessagingStyle/ConversationStyle message. */
 public class LocalImageResolver {
 
+    private static final boolean LOGD = false;
     private static final String TAG = "LocalImageResolver";
 
     /** There's no max size specified, load at original size. */
@@ -236,7 +237,9 @@ public class LocalImageResolver {
         // in some cases despite it not saying so. Rethrow it as an IOException to keep
         // our API contract.
         } catch (IOException | Resources.NotFoundException e) {
-            Log.d(TAG, "Couldn't use ImageDecoder for drawable, falling back to non-resized load.");
+            if (LOGD) {
+                Log.d(TAG, "Couldn't use ImageDecoder for drawable, falling back to non-resized load.");
+            }
             return null;
         }
     }
-- 
2.43.1

